sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -401,9 +411,98 @@ def _get_environment(
         env = self._environment_from_row(row)
         return env
 
+    def _backup_table(self, schema_version: int, table_name: str) -> None:
```

Comment:
Should we also take into account the sqlglot version?

Reply:
I included the sqlglot version in the backup metadata table.